CCXDEV-4898: webconsole insights widget tests

Fewwy opened the pull request.

Commit:
Update i18n keys

Fewwy committed:
first tests to make snapshots

Fewwy committed:
changing the test file to test.tsx

Fewwy committed:
added responses and k8results as props

Fewwy committed:
added responses and k8results as props

Fewwy committed:
added responses and k8results as props

Fewwy committed:
added responses and k8results as props

Fewwy committed:
added responses and k8results as props

Fewwy committed:
added responses and k8results as props

Fewwy committed:
adding fixtures and functions into separate file

Fewwy committed:
adding fixtures and functions into separate file

Fewwy committed:
adding fixtures and functions into separate file

Commit:
Reduces fixtures; creates snapshots

Fewwy committed:
fixed the error and added the mount tests instead of shallow ones

Fewwy committed:
removed redundand code

Fewwy committed:
added test for errors

Fewwy committed:
added tests to check the error and disabled tests and they are failing

gkarat commented on frontend/packages/insights-plugin/__tests__/insights.fixtures.ts:
```diff
@@ -0,0 +1,108 @@
+import { PrometheusHealthPopupProps } from '@console/plugin-sdk';
+
+const funcs = () => ({
```
`funcs` can not be used for the InsightsPopup test (therefore, they are not needed at all). See props (k8sResult, responses) given to this component.

gkarat commented on frontend/packages/insights-plugin/__tests__/insights.fixtures.ts:
```diff
@@ -0,0 +1,108 @@
+  useTranslation: jest.fn(),
+});
+
+const propsForInsightsWidget: PrometheusHealthPopupProps = {
```
nit-pick: rename `propsForInsightsWidget` just to `fixtures` or `propsInsightsPopup`

gkarat commented on frontend/packages/insights-plugin/__tests__/insights.fixtures.ts:
```diff
@@ -0,0 +1,108 @@
+    loaded: true,
+    data: {
+      spec: {
+        clusterID: 'd4b48e95-83e8-436c-b2c9-283c05aecc7f',
```
Please, modify `clusterID` as it seems you have a real ID with some pieces of the real data. This data can't be freely shared, even in PRs.

gkarat commented on frontend/packages/insights-plugin/__tests__/insights.fixtures.ts:
```diff
@@ -0,0 +1,108 @@
+  },
+};
+
+const fixtures = {
```
can be omitted, if you rename `propsForInsightsWidget` to `fixtures` already

gkarat commented on frontend/packages/insights-plugin/__tests__/insights.fixtures.ts:
```diff
@@ -0,0 +1,108 @@
+};
+
+export default fixtures;
+export { funcs };
```
unnecessary

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
```
Try to use `shallow` for all 3 sets of tests instead of `mount`. The tests you've written should work the same if we stick to `shallow`. Try it out

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import { InsightsPopup } from '../src/components/InsightsPopup/index';
+import fixtures, { funcs } from './insights.fixtures';
```
unnecessary funcs

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+import { InsightsPopup } from '../src/components/InsightsPopup/index';
+import fixtures, { funcs } from './insights.fixtures';
+
+describe('InsightsPopup ', () => {
```
nit-pick: a bit more descriptive naming? e.g. InsightsPopup: metrics received state

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+    k8sResult: { ...fixtures.propsForInsightsWidget.k8sResult },
+  };
+
+  const wrapper = mount(<InsightsPopup {...props} />);
```
`mount` -> `shallow`?

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+
+  const wrapper = mount(<InsightsPopup {...props} />);
+  it('main container renders correctly', () => {
+    expect(wrapper.debug()).toMatchSnapshot();
```
if shallowed, debug can be omitted

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+    expect(wrapper.debug()).toMatchSnapshot();
+  });
+  it('should have the correct error component', () => {
+    expect(wrapper.find('.co-insights__box'));
```
it will always result to true. Add any condition after calling `expect`

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+  it('should have the correct error component', () => {
+    expect(wrapper.find('.co-insights__box'));
+  });
+  it('should have correct text inside co-insights__box', () => {
```
```suggestion
  it('should have correct text inside.co-insights__box', () => {
```

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+
+describe('InsightsPopup error state', () => {
+  const functions = funcs();
+  const props = {
```
Make sure, you have to modify responses, so you can do something like
const responses = fixtures.responses; responses[0].response.data.result = [];

gkarat replied:
then you can do 
```suggestion
  const props = {
     ...fixtures,
     responses
  }
```

gkarat commented on frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx:
```diff
@@ -0,0 +1,73 @@
+
+describe('InsightsPopup ', () => {
+  const functions = funcs();
+  const props = {
```
```suggestion
  const props = {
  ...fixtures
  }
```
should be enough

Fewwy committed:
did refactoring and update the code based on georgy comments and suggestions

Fewwy committed:
need to add -1 to the value that insights operator reads but failing

Fewwy committed:
tried to target the second index of the value array and failed

Fewwy committed:
I tried 2 ways to target the value and they look and read worse than this variant. what do you think?

Fewwy committed:
I tried 2 ways to target the value and they look and read worse than this variant. what do you think?

Commit:
Restored unnecessary files from master

Commit:
Refactors unit tests for Insights Plugin

Reduced unnecessary parts of code, added some new tests;
Renamed files associated with tests
